app/controllers/api/v1/users_controller.rb

```diff
@@ -1,34 +1,42 @@
 # frozen_string_literal: true
 
 class Api::V1::UsersController < ApplicationController
-  before_action :check_permissions, only: %i[update destroy]
+  before_action :check_login
 
   # GET /users/1
   def show
+    authorize user
     render json: UserSerializer.render(user)
   end
 
   # POST /users
   def create
     user = User.new(user_params)
+    authorize user
     if user.save
       render json: UserSerializer.render(user), status: :created
-    else
-      render_error(4022, user.errors.messages[:error])
+    elsif email_taken
+      render_error(4010)
```

Comment:
I prefer to use it at the beginning of this method, like `return render_error(4010) if user_by_email` in order to stop running code if there is an error.